src/festim/initial_condition.py

```diff
@@ -66,6 +72,29 @@ def create_expr_fenics(self, mesh, temperature, function_space):
             )
 
 
+class InitialConcentrationFromFile:
+    expr_fenics: fem.Function
+
+    def __init__(self, filename, species, name: str, timestamp: float):
+        self.filename = filename
+        self.name = name
+        self.timestamp = timestamp
+        self.species = species
+
+    def read_function(self, element="P", order=1) -> fem.Function:
+        mesh_in = adios4dolfinx.read_mesh(self.filename, MPI.COMM_WORLD)
+        V_in = fem.functionspace(mesh_in, (element, order))
+        u_in = fem.Function(V_in)
+        adios4dolfinx.read_function(
+            filename=self.filename,
+            u=u_in,
+            name=self.name,
+            time=self.timestamp,
+        )
```

Comment:
Could be worth adding some docs with a link to adios4dolfinx documentation for more information on these functions

Reply:
Done